### PR TITLE
chore(eslint): changed config and fixed eslint problems

### PR DESCRIPTION
**Pull request by e11sy**
## Problem
- Since eslint-config-codex was updated, a lot of eslint problems appeared in repo, unfixable in one PR

## Solution
We should solve problems one by one
- now actual rules that create errors are ignored, they would be returned in next fix PRs
- added tsconfig.eslint.json file to determine include and exclude file paths for eslint
- solved some eslint problems like `prefer-global`, `jsdoc`, `async-function-without-async-parts`, `unsafe-return-any`

### CODE REVIEW

File: codex-ui/src/vue/components/radio/RadioGroup.vue [hunk 1]
```diff
@@ -35,7 +35,7 @@ defineProps<{
   name?: string;
 }>();
 
-const model = defineModel();
+const model = defineModel<String>();
```
Comment (neSpecc) [inline codex-ui/src/vue/components/radio/RadioGroup.vue:38]:
```suggestion
const model = defineModel<string>();
```

File: eslint.config.mjs [hunk 2]
```diff
@@ -1,26 +1,83 @@
...
+    },
+
+    languageOptions: {
+      parser: TsParser,
+      parserOptions: {
+        project: './tsconfig.eslint.json',
+        tsconfigRootDir: './',
+        sourceType: 'module', // Allows for the use of imports
+      },
+    },
```
Comment (neSpecc) [inline eslint.config.mjs:24]: add comment please, why these lines are used for

File: eslint.config.mjs [hunk 3]
```diff
@@ -1,26 +1,83 @@
...
+      '@typescript-eslint/no-unsafe-return': ['off'],
+      '@typescript-eslint/no-unsafe-call': ['off'],
+      '@typescript-eslint/no-unsafe-member-access': ['off'],
+      'jsdoc/require-param-type': ['off'],
+      'jsdoc/informative-docs': ['off'],
+      'jsdoc/require-jsdoc': ['off'],
+    },
+  },
+  {
     languageOptions: {
```
Comment (neSpecc) [inline eslint.config.mjs:58]: add comments for this section as well please

File: src/domain/auth.service.ts [hunk 4]
```diff
@@ -51,13 +51,15 @@ export default class AuthService {
...
-  public async acceptSession(accessToken: string, refreshToken: string): Promise<void> {
+  public acceptSession(accessToken: string, refreshToken: string): void {
     this.eventBus.dispatchEvent(
       new AuthCompletedEvent({
         accessToken,
         refreshToken,
       })
     );
+
+    return;
```
Comment (neSpecc) [inline src/domain/auth.service.ts:62]: seems redundant

File: src/infrastructure/user.repository.ts [hunk 5]
```diff
@@ -44,8 +44,10 @@ export default class UserRepository extends Repository<UserStore, UserStoreData>
   /**
    * Removes user data from the storage
    */
-  public async removeUser(): Promise<void> {
+  public removeUser(): void {
     this.store.removeUser();
+
+    return;
```
Comment (neSpecc) [inline src/infrastructure/user.repository.ts:50]: seems redundant

File: eslint.config.mjs [hunk 6]
```diff
@@ -1,26 +1,83 @@
...
 /**
  * @todo connect architecture config
  */
 export default [
   ...CodeX,
   {
-    name: 'notex.api',
-    files: ['src/**/*'],
+    name: 'ts-notex.web',
+    ignores: ['codex-ui/**/*', '*.pcss', '*.otf', 'eslint.config.mjs', 'postcss.config.js'],
```
Comment (neSpecc) [inline eslint.config.mjs:12]: they are ignored by tsconfig.eslint.ts aren't they?

File: vite.config.ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import { defineConfig, loadEnv } from 'vite';
 import vue from '@vitejs/plugin-vue';
 import hawkVitePlugin from '@hawk.so/vite-plugin';
+import process from 'process';
```
Comment (neSpecc) [inline vite.config.ts:4]: is it used?

File: eslint.config.mjs (hunk 6 above)
Comment (e11sy) [reply eslint.config.mjs:24]: changed tsconfig.eslint.ts

**Review by slaveeks — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]

File: src/presentation/pages/Home.vue [hunk 8]
```diff
@@ -66,6 +66,8 @@ useHead({
  * Returns card subtitle text
  *
  * @param note - Note entity
+ *
+ * @returns {string | undefined}
```
[the remark this answers is not in the document]
Comment (e11sy) [reply src/presentation/pages/Home.vue:70]: made issue in eslint-config codex for that

**Pull request merged**